packages/pluggableWidgets/datagrid-web/tests/e2e/objects/datagrid.widget.ts

```diff
@@ -0,0 +1,10 @@
+class Datagrid {
+    private rows: string[];
+    getAllRows(items: WebdriverIO.ElementArray): string[] {
+        this.rows = [];
+        /* TODO: When I tried to use getText() in a loop to get the text it's taking almost 1min to process. So instead of using getText I am using getHTML with replace for this case. Maybe in the future will be possible to back to getText() */
+        items.map(item => this.rows.push(item.getHTML().replace(/(<([^>]+)>)/gi, "")));
```

Comment:
How about instead of creating an array `this.rows`, return `items.map`? This avoids creating an array by instead mapping over items and returning the resulting array instead.

Reply:
If you agree, you can remove the private property `rows`, too.

Reply:
I'm not a fan of a class being used here, as no state is needed. Instead I suggest creating functions and exporting these.

Reply:
I tried to return the `items.maps` but that will return numbers and I am expecting an array of strings. I changed to use a const instead of `this.rows` and removed the private property `rows`.
For the declared class basically I'm following the pattern which we're using in other specs. So, I think we can keep that way. What do you think?

Reply:
I'm all for conventions so keep it the same (= class).

As for the function body, it can be reduced to: `items.map(item => item.getHTML().replace(/(<([^>]+)>)/gi, ""));` which can be returned immediately in the function. The reason it was returning numbers it because `rows.push()` returns a number.

Reply:
Thanks! Fixed!